yt_dlp/extractor/megatvcom.py

```diff
@@ -0,0 +1,177 @@
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'display_id': display_id,
+            'title': title,
+            'description': description,
+            'thumbnail': thumbnail,
+            'timestamp': timestamp,
+            'formats': formats,
+            'subtitles': subs,
+        }
+
+
+class MegaTVComEmbedIE(MegaTVComBaseIE):
+    IE_NAME = 'megatvcom:embed'
+    IE_DESC = 'megatv.com embedded videos'
+    _VALID_URL = r'(?:https?:)?//(?:www\.)?megatv\.com/embed/?\?p=(?P<id>\d+)'
+    _EMBED_RE = re.compile(rf'''<iframe[^>]+?src=(?P<_q1>["'])(?P<url>{_VALID_URL})(?P=_q1)''')
+
+    _TESTS = [{
+        'url': 'https://www.megatv.com/embed/?p=2020520979',
+        'md5': '6546a1a37fff0dd51c9dce5f490b7d7d',
+        'info_dict': {
+            'id': '520979',
+            'ext': 'mp4',
+            'title': 'md5:70eef71a9cd2c1ecff7ee428354dded2',
+            'description': 'md5:0209fa8d318128569c0d256a5c404db1',
+            'timestamp': 1634975747,
+            'upload_date': '20211023',
+            'display_id': 'egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia',
+            'thumbnail': 'https://www.megatv.com/wp-content/uploads/2021/10/ΠΕΙΡΑΙΑΣ-1024x450.jpg',
+        },
+    }, {
+        'url': 'https://www.megatv.com/embed/?p=2020534081',
+        'md5': '6ac8b3ce4dc6120c802f780a1e6b3812',
+        'info_dict': {
+            'id': '534081',
+            'ext': 'mp4',
+            'title': 'md5:062e9d5976ef854d8bdc1f5724d9b2d0',
+            'description': 'md5:36dbe4c3762d2ede9513eea8d07f6d52',
+            'timestamp': 1636376351,
+            'upload_date': '20211108',
+            'display_id': 'neo-rekor-stin-timi-tou-ilektrikou-reymatos-pano-apo-ta-200e-i-xondriki-timi-tou-ilektrikou',
+            'thumbnail': 'https://www.megatv.com/wp-content/uploads/2021/11/Capture-266.jpg',
+        },
+    }]
+
+    @classmethod
+    def _extract_urls(cls, webpage):
+        for mobj in cls._EMBED_RE.finditer(webpage):
+            yield unescapeHTML(mobj.group('url'))
+
+    def _match_canonical_url(self, webpage):
+        LINK_RE = r'''(?x)
+        <link(?:
+            rel=(?P<_q1>["'])(?P<canonical>canonical)(?P=_q1)|
+            href=(?P<_q2>["'])(?P<href>(?:(?!(?P=_q2)).)+)(?P=_q2)|
+            [^>]*?
+        )+>
+        '''
+        for mobj in re.finditer(LINK_RE, webpage):
+            canonical, href = mobj.group('canonical', 'href')
+            if canonical and href:
+                return unescapeHTML(href)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        player_attrs = self._extract_player_attrs(webpage)
+        canonical_url = player_attrs.get('share_url') or self._match_canonical_url(webpage)
+        if not canonical_url:
+            raise ExtractorError('canonical URL not found')
+        video_id = parse_qs(urlparse(canonical_url).query)['p'][0]
```

Comment:
```suggestion
        video_id = parse_qs(canonical_url)['p'][0]
```
import `parse_qs` from utils